input/input.go

```diff
@@ -0,0 +1,252 @@
+	// checking if at least one job is present.
+	if len(c.AuditJobs) == 0 {
+		return ErrMissingAuditJob
+	}
+	// checking if at least one target is present.
+	if len(c.Targets) == 0 {
+		return ErrMissingTarget
+	}
+	for _, j := range c.AuditJobs {
+		// checking if audit job name is not empty.
+		if j.Name == "" {
+			return ErrMissingAuditJobName
+		}
+		// checking if access token is not empty.
+		if j.AccessToken == "" {
+			return ErrMissingAccessToken
+		}
+		// checking if either username or email is not empty.
+		if j.Username == "" && j.Email == "" {
+			return ErrMissingUsernameEmail
+		}
+		// checking if repository name is not empty.
+		if j.RepositoryName == "" {
+			return ErrMissingRepositoryName
+		}
+		// checking if repository url is not empty.
+		if j.RepositoryURL == "" {
+			return ErrMissingRepositoryURL
+		}
+		// checking if repository type is not empty.
+		if j.RepositoryType == "" {
+			return ErrMissingRepositoryType
+		}
+		// checking if any target is defined in output.
+		if len(j.TargetName) == 0 {
+			return ErrMissingTargetName
+		}
+		// checking if polling interval is not empty.
+		if j.PollingInterval == "" {
+			return ErrMissingPollingInterval
+		}
+		if j.PollingInterval != "" {
+			return checkPollingIntervalFormat(j.PollingInterval)
+		}
+	}
+	for _, t := range c.Targets {
+		// checking if target name is missing.
+		if t.Name == "" {
+			return ErrMissingTargetName
+		}
+		// checking if target type is missing.
+		if t.Type == "" {
+			return ErrMissingTargetType
+		}
+	}
+	return nil
+}
+
+// populateDefaultValues puts default values to optional fields in config.
+func (c *Config) populateDefaultValues() {
+	for i := range c.AuditJobs {
+		if len(c.AuditJobs[i].Branches) == 0 {
+			c.AuditJobs[i].Branches = append(c.AuditJobs[i].Branches, DefaultBranch)
+		}
+		//populating access token from environment variable if present. environment variable name is same as auditjob name.
+		accessTokenFromEnv := os.Getenv(c.AuditJobs[i].Name)
+		if accessTokenFromEnv != "" {
+			c.AuditJobs[i].AccessToken = accessTokenFromEnv
+		}
+	}
+}
+
+/*
+checkPollingIntervalFormat checks the format for polling interval.
+Format: Integer[s/m/h/d]  (s:seconds , m:minutes , h:hours , d:days).
+Example: Accepted formats are 10s, 10m, 10h, 1d
+*/
+func checkPollingIntervalFormat(pi string) error {
+	if pi == "" {
+		return ErrMissingPollingInterval
+	}
+	lastChar := string(pi[len(pi)-1])
+	if lastChar == "h" || lastChar == "s" || lastChar == "m" || lastChar == "d" {
+		remChar := string(pi[0 : len(pi)-1])
+		if !isNumeric(remChar) {
+			return ErrPollingIntervalFormat
+		}
+	} else {
+		return ErrPollingIntervalFormat
+	}
+	return nil
+}
+
+//isNumeric checks if string is numeric or not
+func isNumeric(s string) bool {
+	_, err := strconv.ParseFloat(s, 64)
+	return err == nil
+}
+
+// AuditJob represents the auditing job for which data will be fetched from remote git repository.
+type AuditJob struct {
```

Comment:
Can you put structs in the begining of the files, then add functions

Reply:
changed